jwst/ami/oifits.py

```diff
@@ -41,44 +42,208 @@ def __init__(self, fringefitter, method="median"):
         self.ctrs_inst = self.fringe_fitter.instrument_data.ctrs_inst
-        self.pa = (
-            self.fringe_fitter.instrument_data.pav3
-        )  # header pav3, not including v3i_yang??
 
         self.bholes, self.bls = self._makebaselines()
         self.tholes, self.tuv = self._maketriples_all()
+        self.qholes, self.quads = self._makequads_all()
 
     def make_obsarrays(self):
         """
         Make arrays of observables of the correct shape for saving to datamodels
         """
-        # arrays of observables, (nslices,nobservables) shape.
+        # empty arrays of observables, (nslices,nobservables) shape.
         self.fringe_phases = np.zeros((self.nslices, self.n_baselines))
         self.fringe_amplitudes = np.zeros((self.nslices, self.n_baselines))
         self.closure_phases = np.zeros((self.nslices, self.n_closure_phases))
+        self.t3_amplitudes = np.zeros((self.nslices, self.n_closure_phases))
+        self.q4_phases = np.zeros((self.nslices, self.n_closure_amplitudes))
         self.closure_amplitudes = np.zeros((self.nslices, self.n_closure_amplitudes))
         self.pistons = np.zeros((self.nslices, self.n_holes))
         # model parameters
         self.solns = np.zeros((self.nslices, 44))
 
+        # populate with each integration's observables
         for i, nrmslc in enumerate(self.fringe_fitter.nrm_list):
-            self.fringe_phases[i, :] = np.rad2deg(nrmslc.fringephase)  # FPs in degrees
+            self.fringe_phases[i, :] = nrmslc.fringephase  # FPs in radians
             self.fringe_amplitudes[i, :] = nrmslc.fringeamp
-            self.closure_phases[i, :] = np.rad2deg(nrmslc.redundant_cps)  # CPs in degrees
+            self.closure_phases[i, :] = nrmslc.redundant_cps  # CPs in radians
+            self.t3_amplitudes[i, :] = nrmslc.t3_amplitudes
+            self.q4_phases[i, :] = nrmslc.q4_phases # quad phases in radians
             self.closure_amplitudes[i, :] = nrmslc.redundant_cas
-            self.pistons[i, :] = np.rad2deg(
-                nrmslc.fringepistons
-            )  # segment pistons in degrees
+            self.pistons[i, :] = nrmslc.fringepistons # segment pistons in radians
             self.solns[i, :] = nrmslc.soln
 
         self.fringe_amplitudes_squared = self.fringe_amplitudes ** 2  # squared visibilities
 
+    def rotate_matrix(self, cov_mat, theta):
+        """
+        Rotate a covariance matrix by an angle.
+
+        Parameters
+        ----------
+        cov_mat:
+        theta: float
+            Angle by which to rotate the matrix (radians)
+
+        Returns
+        -------
+
+        """
+        c, s = np.cos(theta), np.sin(theta)
+        R_mat = [[c, -s],
+                 [s, c]]
+        # coordinate rotation from real/imaginary to absolute value/phase (modulus/argument)
+        cv_rotated = np.linalg.multi_dot([np.transpose(R_mat), cov_mat, R_mat])
+        return cv_rotated
```

Comment:
Variables should have lower case names wherever possible
```suggestion
        r_mat = [[c, -s],
                 [s, c]]
        # coordinate rotation from real/imaginary to absolute value/phase (modulus/argument)
        cv_rotated = np.linalg.multi_dot([np.transpose(r_mat), cov_mat, r_mat])
        return cv_rotated
```